mycroft/audio/services/guiplayer/__init__.py

```diff
@@ -0,0 +1,211 @@
+#
+
+import time
+import mimetypes
+import re
+
+from threading import Lock
+from mycroft.audio.services import AudioBackend
+from mycroft.messagebus.message import Message
+from mycroft.util.log import LOG
+
+
+class GuiPlayerService(AudioBackend):
+    def __init__(self, config, bus=None, name='guiplayer'):
+        super(GuiPlayerService, self).__init__(config, bus)
+        self.config = config
+        self.process = None
+        self.bus = bus
+        self.name = name
+        self._is_playing = False
+        self._paused = False
+        self.tracks = []
+        self.index = 0
+        self.track_lock = Lock()
+        self.track_meta_from_player = None
+        self.track_meta_from_cps = None
+
+        self.bus.on('GuiPlayerServicePlay', self._play)
+        self.bus.on(
+            'gui.player.media.service.sync.status',
+            self.track_media_status)
+        self.bus.on(
+            'gui.player.media.service.get.meta',
+            self.sync_meta_from_player)
+        self.bus.on('play:status', self.sync_meta_to_player)
+
+    def supported_uris(self):
+        return ['file', 'http', 'https']
+
+    def clear_list(self):
+        with self.track_lock:
+            self.tracks = []
+
+    def add_list(self, tracks):
+        with self.track_lock:
+            self.tracks += tracks
+            LOG.info("Track list is " + str(tracks))
+
+    def _get_track(self, track_data):
+        if isinstance(track_data, list):
+            track = track_data[0]
+            mime = track_data[1]
+            mime = mime.split('/')
+        else:  # Assume string
+            track = track_data
+            mime = self.find_mime(track)
+        return track, mime
+
+    def _play(self, message):
+        """ Handle _play call from play signal. """
+        repeat = message.data.get("repeat", False)
+        with self.track_lock:
+            if len(self.tracks) > self.index:
+                track, mime = self._get_track(self.tracks[self.index])
+            else:
+                return
+
+        # Indicate to audio service which track is being played
+        if self._track_start_callback:
+            self._track_start_callback(track)
+
+        try:
+            if 'video' in mime[0]:
+                LOG.debug("Sending Video Type")
+                self.bus.emit(Message("playback.display.video.type"))
+            else:
+                LOG.debug("Sending Audio Type")
+                self.bus.emit(Message("playback.display.audio.type"))
+        except BaseException:
+            LOG.debug("Cannot Determine Mime Type Falling Back To List Check")
+            mediatype = self.fallback_type_check(track)
+            if mediatype == "video":
+                LOG.debug("Falling Back To Audio Type")
+                self.bus.emit(Message("playback.display.video.type"))
+            else:
+                LOG.debug("Falling Back To Audio Type")
+                self.bus.emit(Message("playback.display.audio.type"))
```

Comment:
I'm not sure what's happening, but using a slightly modified version of Ake's YT Skill, I'm seeing this log message that its falling back to Audio but it then uses the Video player.

I added logs to the `handle_display_video` and `handle_display_audio` methods in the control Skill and it somehow is triggering the Video handler.

Reply:
What type of stream is Ake's YT Skill providing ? Video or Audio ?

Reply:
The debug message was wrong and has been fixed in newer commit

Reply:
If Ake's YT stream URL consist of "videoplayback" in the URL it will fallback to Video type as that is the only sane and logical way to check if the mime type for those type of streaming links is actually an Video.